src/helpers/is-metadata-json.validator.ts

```diff
@@ -0,0 +1,56 @@
+import {
+    registerDecorator,
+    ValidationArguments,
+    ValidationOptions,
+    ValidatorConstraint,
+    ValidatorConstraintInterface,
+} from "class-validator";
+
+@ValidatorConstraint({ name: "isMetadataJson", async: false })
+export class IsMetadataJsonConstraint implements ValidatorConstraintInterface {
+    private message = "";
+
+    public validate(value: string, args: ValidationArguments): boolean {
+        const [propertyName] = args.constraints;
+        this.message = `${propertyName} must be a valid list of metadata keys and values`;
+
+        if (typeof value !== "string") {
+            return false;
+        }
+        try {
+            const json = JSON.parse(value) as Record<string, string>;
+
+            for (const key of Object.keys(json)) {
+                if (typeof key !== "string" || key.trim() === "") {
+                    this.message = `The key whose value is "${json[key]}" must be a valid text value`;
```

Comment:
Use proper translation key (+ translation in frontend)

Reply:
Agreed. I've done away with hinting at which row has the error. The current setup (i.e. throwing a generic BadRequestException in nest.js) doesn't easily allow additional info to be passed to the frontend. That, or I'm blind.

Reply:
In case you have a suggestion, I'll leave this open. Otherwise, you can merge this PR

Reply:
Translation key is fine.